src/main/java/org/traccar/protocol/LaipacProtocolDecoder.java

```diff
@@ -164,4 +268,46 @@ protected Object decode(
         return position;
     }
 
+    @Override
+    protected Object decode(
+        Channel channel, SocketAddress remoteAddress, Object msg) throws Exception {
+
+        String sentence = (String) msg;
+
+        // Validate sentence length
+        int slen = sentence.length();
+        if (slen <= 5) {
+            return null;
+        }
+
+        // Validate sentence format
+        if (sentence.charAt(0) != '$') {
+            return null;
+        }
+        if (sentence.charAt(slen - 3) != '*') {
```

Comment:
Why do we need all this validation? That's the whole point of regex.

Reply:
This validation ensures that substring indexes in the code bellow it, does not go out of bounds. Regular expressions are being used to parse specific kind of sentences or frames, without taking cheksums into account. If you don't agree, just tell me, I'll remove the code. Thanks.

Reply:
Same, please remove validation.

Reply:
Ok. Thanks.